lib/maths/CTimeSeriesDecompositionDetail.cc

```diff
@@ -502,14 +623,16 @@ bool CTimeSeriesDecompositionDetail::CPeriodicityTest::acceptRestoreTraverser(
                 traverser.traverseSubLevel([this](core::CStateRestoreTraverser& traverser_) {
                     return m_Machine.acceptRestoreTraverser(traverser_);
                 }))
+        RESTORE_NO_ERROR(SHORT_WINDOW_6_3_TAG, m_Windows[E_Short] = this->newWindow(E_Short))
         RESTORE_SETUP_TEARDOWN(
-            SHORT_WINDOW_6_3_TAG, m_Windows[E_Short].reset(this->newWindow(E_Short)),
+            SHORT_WINDOW_7_9_TAG, m_Windows[E_Short] = this->newWindow(E_Short),
             m_Windows[E_Short] && traverser.traverseSubLevel(std::bind(
                                       &CExpandingWindow::acceptRestoreTraverser,
                                       m_Windows[E_Short].get(), std::placeholders::_1)),
             /**/)
+        RESTORE_NO_ERROR(LONG_WINDOW_6_3_TAG, m_Windows[E_Long] = this->newWindow(E_Long))
         RESTORE_SETUP_TEARDOWN(
-            LONG_WINDOW_6_3_TAG, m_Windows[E_Long].reset(this->newWindow(E_Long)),
+            LONG_WINDOW_7_9_TAG, m_Windows[E_Long] = this->newWindow(E_Long),
```

Comment:
Short and long windows from old versions are ignored, but the format hasn't changed.  So is the intention just to reinitialise from scratch when 7.9 is used for the first time because we think the persisted windows are sub-optimal?  Will this have any impact on anomaly detection?  Will every user who upgrades to 7.9 see strange effects in the model plot immediately after upgrading?

It would be good to have comments here to say what the intention is.

Reply:
This is intentional because otherwise we'll just use the old poor parameterisation for existing jobs. Since the window slides (after it has fully expanded) this just means we loose the temporary state we'd accumulated in the window to the current time, so this seemed worthwhile.

Reply:
See [93cbad3](https://github.com/elastic/ml-cpp/pull/1297/commits/93cbad37cbef56122803d5743d9ef6989eef8fd0).